core/src/Cabana_Parallel.hpp

```diff
@@ -47,6 +56,16 @@ KOKKOS_FORCEINLINE_FUNCTION
     functor( t, s, a );
 }
 
+template <class WorkTag, class FunctorType, class IndexType>
+KOKKOS_FORCEINLINE_FUNCTION
+    typename std::enable_if<!std::is_same<WorkTag, void>::value>::type
+    functorTagDispatch( const FunctorType &functor, const IndexType i,
+                        const IndexType j, const IndexType k )
+{
+    const WorkTag t{};
+    functor( t, i, j, k );
```

Comment:
you can also do the generic index forwarding here

Reply:
done